brilift/src/main.rs

```diff
@@ -583,7 +606,7 @@ impl<M: Module> Translator<M> {
         }
 
         // Insert instructions.
-        compile_body(&func.instrs, &mut builder, &env);
+        env.compile_body(&func.instrs, &mut builder);
```

Comment:
Moving `compile_body` and friends to an `impl` for `CompileEnv` is a good organizational idea!